src/hooks/useSubmitButtonVisibility.ts

```diff
@@ -0,0 +1,58 @@
+import {useEffect, useRef, useState} from 'react';
+import {Dimensions} from 'react-native';
+import useSafeAreaInsets from './useSafeAreaInsets';
+import useThemeStyles from './useThemeStyles';
+import useWindowDimensions from './useWindowDimensions';
+
+// Useful when there's a need to hide the submit button from FormProvider,
+// to let form content fill the page when virtual keyboard is shown
+function useSubmitButtonVisibility() {
+    const styles = useThemeStyles();
+    const {windowHeight, isSmallScreenWidth} = useWindowDimensions();
+    const [isSubmitButtonVisible, setIsSubmitButtonVisible] = useState(true);
+    const initialWindowHeightRef = useRef(windowHeight);
+    const isSmallScreenWidthRef = useRef(isSmallScreenWidth);
+    const {bottom} = useSafeAreaInsets();
+
+    // Web: the submit button is shown when the height of the window is the same or greater,
+    // otherwise it's hidden
+    useEffect(() => {
+        const dimensionsListener = Dimensions.addEventListener('change', ({window}) => {
```

Comment:
I think I understand what we're trying to do with this functionality (making more space when users shrink the window), but what inspired it? Seems kind of overkill to me on first glance.

Reply:
On mWeb, there's no API to detect if the keyboard is shown. So I created this logic to detect if the user is running the web app in mobile by checking if the screen orientation is `portrait` and the `platform === 'web'`. When the keyboard is shown the screen height shrinks, so this listener check for that change and hides the submit button when that occurs, and show it again when the current screen height equals the initial screen height